# Dataset Overview

In [1]:
# import the required libraries
import pandas as pd
import numpy as np
from sklearn import linear_model

In [2]:
# read the dataset
df=pd.read_excel("C:\\Users\\abc\\Documents\\hiring.xlsx")

In [3]:
# print the dataset
df

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0        NaN                    8.0                           9      50000
1        NaN                    8.0                           6      45000
2      five                     6.0                           7      60000
3        two                   10.0                          10      65000
4     seven                     9.0                           6      70000
5      three                    7.0                          10      62000
6        ten                    NaN                           7      72000
7     eleven                    7.0                           8      80000

In [4]:
# number of rows and columns in the datatset
df.shape

(8, 4)

In [5]:
# name of all the columns of the dataset
for n in df.columns:
    print(n)

experience
test_score(out of 10)
interview_score(out of 10)
salary($)


In [6]:
# basic details about the dataset
df.describe()

test_score(out of 10)  interview_score(out of 10)    salary($)
count               7.000000                    8.000000      8.00000
mean                7.857143                    7.875000  63000.00000
std                 1.345185                    1.642081  11501.55269
min                 6.000000                    6.000000  45000.00000
25%                 7.000000                    6.750000  57500.00000
50%                 8.000000                    7.500000  63500.00000
75%                 8.500000                    9.250000  70500.00000
max                10.000000                   10.000000  80000.00000

In [7]:
# datatypes of the columns dataset
df.dtypes

experience                     object
test_score(out of 10)         float64
interview_score(out of 10)      int64
salary($)                       int64
dtype: object

# Data Preprocessing

In [8]:
# all the null values in the all the columns of the dataset
df.isnull().sum()

experience                    2
test_score(out of 10)         1
interview_score(out of 10)    0
salary($)                     0
dtype: int64

In [9]:
# fill null values in experience column with 'zero'
df.experience=df.experience.fillna('zero')

In [10]:
# extract experience column of the dataset
df.experience

0      zero
1      zero
2     five 
3       two
4    seven 
5     three
6       ten
7    eleven
Name: experience, dtype: object

In [11]:
# create a variable data
data = np.array([0, 0,5,2,7,3,10,11])

In [12]:
# print data variable
data

array([ 0,  0,  5,  2,  7,  3, 10, 11])

In [13]:
# create a ser variable
ser = pd.Series(data)

In [14]:
# print ser variable
ser

0     0
1     0
2     5
3     2
4     7
5     3
6    10
7    11
dtype: int32

In [15]:
# assign ser in df['experience']
df['experience']=ser

In [16]:
# print df['experience']
df['experience']

0     0
1     0
2     5
3     2
4     7
5     3
6    10
7    11
Name: experience, dtype: int32

In [17]:
# print df
df

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0           0                    8.0                           9      50000
1           0                    8.0                           6      45000
2           5                    6.0                           7      60000
3           2                   10.0                          10      65000
4           7                    9.0                           6      70000
5           3                    7.0                          10      62000
6          10                    NaN                           7      72000
7          11                    7.0                           8      80000

In [18]:
# data types of all the columns
df.dtypes

experience                      int32
test_score(out of 10)         float64
interview_score(out of 10)      int64
salary($)                       int64
dtype: object

In [19]:
# create a variable a 
a=df['test_score(out of 10)'].mean()

In [20]:
# print variable a
a

7.857142857142857

In [21]:
# fill null values in test_score(out of 10) column with variable a
df['test_score(out of 10)']=df['test_score(out of 10)'].fillna(a)

In [22]:
# extract test_score(out of 10) column from the dataset
df['test_score(out of 10)']

0     8.000000
1     8.000000
2     6.000000
3    10.000000
4     9.000000
5     7.000000
6     7.857143
7     7.000000
Name: test_score(out of 10), dtype: float64

In [23]:
# convert data type from float to int
df['test_score(out of 10)']=df['test_score(out of 10)'].astype(int)

In [24]:
# extract the column test_score(out of 10) from the dataset
df['test_score(out of 10)']

0     8
1     8
2     6
3    10
4     9
5     7
6     7
7     7
Name: test_score(out of 10), dtype: int32

In [25]:
# number of rows and columns of the datatset
df.shape

(8, 4)

In [26]:
# basic information about the dataset
df.describe()

experience  test_score(out of 10)  interview_score(out of 10)  \
count     8.00000                8.00000                    8.000000   
mean      4.75000                7.75000                    7.875000   
std       4.26782                1.28174                    1.642081   
min       0.00000                6.00000                    6.000000   
25%       1.50000                7.00000                    6.750000   
50%       4.00000                7.50000                    7.500000   
75%       7.75000                8.25000                    9.250000   
max      11.00000               10.00000                   10.000000   

         salary($)  
count      8.00000  
mean   63000.00000  
std    11501.55269  
min    45000.00000  
25%    57500.00000  
50%    63500.00000  
75%    70500.00000  
max    80000.00000

In [27]:
# data types of all the columns of the datase
df.dtypes

experience                    int32
test_score(out of 10)         int32
interview_score(out of 10)    int64
salary($)                     int64
dtype: object

# Predict the salary of the candidate if he have 2 years of experience, scored 9 marks in test score and 6 marks in interview score ?

In [28]:
# initialize the linear regression model
reg=linear_model.LinearRegression()

In [29]:
# print reg variable
reg

LinearRegression()

In [45]:
# extract the experience column, and make it dataset
df[['experience']]

experience
0           0
1           0
2           5
3           2
4           7
5           3
6          10
7          11

In [46]:
# extract the test_score(out of 10) column, and make it dataset
df[['test_score(out of 10)']]

test_score(out of 10)
0                      8
1                      8
2                      6
3                     10
4                      9
5                      7
6                      7
7                      7

In [47]:
# extarct the interview_score(out of 10) column, and make it dataset
df[['interview_score(out of 10)']]

interview_score(out of 10)
0                           9
1                           6
2                           7
3                          10
4                           6
5                          10
6                           7
7                           8

In [44]:
# make the numpy array of experience, test_score(out of 10) and interview_score(out of 10) datasets
df[['experience','test_score(out of 10)','interview_score(out of 10)']].values

array([[ 0,  8,  9],
       [ 0,  8,  6],
       [ 5,  6,  7],
       [ 2, 10, 10],
       [ 7,  9,  6],
       [ 3,  7, 10],
       [10,  7,  7],
       [11,  7,  8]], dtype=int64)

In [49]:
# Lets fit the model
reg.fit(df[['experience','test_score(out of 10)','interview_score(out of 10)']].values,df[['salary($)']].values)

LinearRegression()

In [50]:
# lets predict
reg.predict([[2,9,6]])

array([[53713.86677124]])

Hence, the salary of the candidate will be 53713.86677124($)

# Predict the salary of the candidate if he have 12 years of experience, scored 10 marks in test score and 10 marks in interview score ?

In [52]:
# lets predict
reg.predict([[12,10,10]])

array([[93747.79628651]])

Hence, the salary of the candidate will be 93747.79628651($)